src/CDSAPI.jl

```diff
@@ -1,27 +1,32 @@
 module CDSAPI
 
-export py2ju, retrieve
+using HTTP
+using JSON
+using Base64
 
-using HTTP, JSON, Base64
+export
+    py2ju
 
 function retrieve(name, params, filename)
-    cred = Dict()
-    open(joinpath(homedir(), ".cdsapirc"), "r") do f
-        dicttxt = readlines(f)
-        for i in dicttxt
-            tmp = split(i, ":"; limit=2)
-            cred[tmp[1]] = lstrip(tmp[2])
-        end
+    creds = Dict()
+    lines = readlines(open(joinpath(homedir(), ".cdsapirc"), "r"))
```

Comment:
Let's use a do block to make sure the file is closed:

```julia
open(joinpath(homedir(),".cdsapirc")) do f
  cred = Dict()
  for line in readlines(f)
    key, val = strip.(split(line,':'))
    cred[key] = val
  end
end
```

Also, I wonder if we could exploit `JSON.parse` here as well? Is the credential file a JSON too?

Reply:
I tried `JSON.parse` already, but it is not able to parse the keys due to two `:`s (first - `"url : "` and second in - `"https://"`). Same goes for api key too (the second line in .cdsapirc)